gui_agents/store/registry.py

```
@@ -4,19 +4,125 @@
+from typing import Optional
+
+
 class Registry:
-    _services: dict[str, object] = {}
+    """
+    Registry class that supports both global singleton and task-specific instances.
+
+    For backward compatibility, maintains the global singleton behavior while
+    supporting task-level isolation for concurrent execution.
+    """
+    _global_services: dict[str, object] = {}
```

Ruff (0.14.0)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)